### DISCUSSION TITLE
no ssl certificate

### DISCUSSION THREAD

**Discussion by TommieWaters** [debauchee/barrier#2000] [Q&A] [answered]
installed on windows and linux both logs are saying they cant find ssl certificates the folders there looking in are empty. ERROR: ssl certificate doesn't exist: C:\Users\Tommi\AppData\Local\Barrier\SSL\Barrier.pem
 ERROR: could not load client certificates

**Comment by akinnee** [comment]
Same here. I unchecked Enable SSL on both the client and server for now, and things work. Are we expected to generate our own certs for SSL to work?

**Comment by TommieWaters** [comment]
yeah i think so. i have ssl disabled on both too and its still not working and looking for the file in the logs

**Reply by fredodupoux** [reply]
Had the same issue but you have to make sure you reload the server after unchecking Enable SSL it should work.

**Comment by itsrajadarsh** [comment, marked as answer]
You can also use with SSL enabled,
just you need yo create Barrier.pem file at that location:

in linux, follow these commenda:
$ cd /home/adarsh/.local/share/barrier/SSL/
$ mkdir -p Fingerprints
$ openssl req -x509 -nodes -days 365 -subj /CN=barrier -newkey rsa:4096 -keyout Barrier.pem -out Barrier.pem
$ openssl x509 -fingerprint -sha256 -noout -in Barrier.pem > Fingerprints/Local.txt
$ sed -e "s/.*=/v2:sha256:/" -i Fingerprints/Local.txt


and reStart the server

**Reply by TommieWaters** [reply]
yeah this is what i ended up doing eventually. completely forgot about this question i asked. thanks for the reply might help someone else as well.

**Comment by nbolton** [comment]
Please try Deskflow or Input Leap as Barrier is no longer in development.

https://github.com/deskflow/deskflow
https://github.com/input-leap/input-leap

Both are compatible with Barrier client/server so you can try on one of your computers.

If this is still an issue in those projects, we would appreciate a cross-post of this issue.

Not sure which to use? This depends on your needs, but this comparison may help:
https://github.com/deskflow/deskflow/wiki/Project-Forks

You can also check out each project's 'Insights' tab, which may help you decide.

**Reply by TommieWaters** [reply]
had a quick look. thanks for the insight. id probably use deskflow as i need to connect to a raspberry pi as well but interesting there's a new fork that's maintained. ill probably just use barrier for now im kind of a set and forget type person if it works it works. appreciate your response.

**Reply by nbolton** [reply]
No problem, "if it ain't broke" :+1: By the way, technically Deskflow is not a fork, as it's now the top-most upstream repo. Input Leap, Barrier, and Synergy are all downstream of Deskflow.

![image](https://github.com/user-attachments/assets/1bcd92ac-bf35-4f00-b6ba-a3d5434ce8bf)
